lib/jekyll-minifier.rb

```diff
@@ -78,7 +78,22 @@ def write(dest)
       if exclude?(dest, dest_path)
         output_file(dest_path, output)
       else
-        output_html(dest_path, output)
+        case File.extname(dest_path)
```

Comment:
Perhaps this should be broken into a new method. DRY

Reply:
i'm only a few weeks into ruby code, please go ahead and make the changes or suggest them so I can go ahead and do it.

Reply:
what if we did the following inside of the Compressor method:
```ruby
def output_compressed(path, content)
	  case File.extname(path)
		  when '.js'
			if path =~ /.min.js$/
              output_file(path, content)
			else
			  output_js(path, content)
			end
		  when '.css'
			if path =~ /.min.css$/
              output_file(path, content)
			else
			  output_css(path, content)
			end
          when '.xml'
            output_html(path, content)
		  else
            output_html(path, content)
		end
	end
```

Then change up the two file types I updated in this PR to us it of the 'else'.

Reply:
Its your PR I don't think I can. You can add something like: 

```ruby
def output_compressed(path, context)
  case File.extname(path)
  when '.js'
    if path.end_with?('.min.js')
      output_file(path, context)
    else
      output_js(path, context)
    end
  when '.css'
    if path.end_with?('.min.css')
      output_file(path, context)
    else
      output_css(path, context)
    end
  else
    output_html(path, context)
  end
end
```

and reference it by: 
```ruby
output_compressed(dest_path, output
```

Which I think is pretty much what you put above. Avoid `Regex`, its slow, complicated, and easily messed up.